_posts/2017-07-05-jenkins-job-builder.md

```diff
@@ -0,0 +1,125 @@
+project out of my university that wanted to use continuous integration as a
+part of their workflow. While services like [Travis](https://travis-ci.org/)
+and [Semaphore](https://semaphoreci.com/) can be
+excellent if you can accomplish your build within the constraints of the
+systems they provide, they could not meet the needs of that project. Enter
+[Jenkins](https://jenkins.io/), "the leading open source automation server".
+
+Jenkins is an powerful tool, and much of this power comes from the fact that it
+is so incredibly customizable. Configuring jobs via the web UI that Jenkins
+provides leaves many things to be desired, however. First, the task is fraught
+with human error. It is very easy to forget to expand that one nested menu and
+set a critical value, perhaps creating a faulty job that leads to misleading
+results or a broken CI/CD pipeline causing a headache for all involved, slowing
+work down instead of speeding it up (that's what automation is _supposed_ to
+do, right?). Secondly, once you have constructed your house of cards, you have
+only to experience a hard drive failure to lose all your work, or much more
+likely, something changes in your set up that leads you to hand updating dozens
+of jobs so carefully configured with hours of mouse clicks.
+
+Jenkins Job Builder (JJB)
+----------------
+
+To scale the process of Jenkins job definition it become obvious that one must
+be able to generate Jenkins jobs based on templates and with variable
+replacement so that you might test multiple versions against multiple OS's, and
+more over that these assets should be under version control. The good news is,
+this tool already exists!
+
+[Jenkins Job Builder](https://docs.openstack.org/infra/jenkins-job-builder/),
+installable via [pip](https://pypi.python.org/pypi/pip) allows the user to
+generate the job definitions from YAML files and upload the XML job definitions
+to your Jenkins instance.
+
+PulpQE stores the YAML files and scripts used to generate the pulp jenkins jobs
+with jenkins-job-builder in the
+[pulp_packaging](https://github.com/pulp/pulp_packaging) repo. 
+
+While working on changes to these files, it is desirable to be able to try
+them out locally, so I decided to give it a shot!
+
+I have [Docker](https://www.docker.com/) set up on my Fedora 25 machine, so I spun up a Jenkins instance to
+test it out: There is an official [Jenkins image on
+DockerHub](https://hub.docker.com/_/jenkins/) so I pulled it down and served it
+up on my localhost. _(More detailed guides for setting up jenkins from this
+docker image are available on the Jenkins DockerHub page)_
+
+    docker pull jenkins
+    docker run -p 8080:8080 -p 50000:50000 jenkins
+
+I then visited the Jenkins web ui at <code>http://localhost:8080</code> and was directed to
+login with an admin account created during the install with a temporary
+password created printed to the logs. I created a new admin account and password and chose to
```

Comment:
A bit of a run-on sentence. How about:

I then visited the Jenkins web UI at <code>http://localhost:8080</code> and was directed to login with an admin account created during the install. A temporary admin password could be found in the logs. 

Also, I'd probably drop "and password" here too. Too many ands.